packages/react/src/reactrouterv4v5/integration.ts

```diff
@@ -0,0 +1,158 @@
+import { WINDOW, startBrowserTracingNavigationSpan, startBrowserTracingPageLoadSpan } from '@sentry/browser';
+import {
+  SEMANTIC_ATTRIBUTE_SENTRY_OP,
+  SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN,
+  SEMANTIC_ATTRIBUTE_SENTRY_SOURCE,
+  defineIntegration,
+} from '@sentry/core';
+import type { Client, IntegrationFn, TransactionSource } from '@sentry/types';
+import { logger } from '@sentry/utils';
+import { DEBUG_BUILD } from '../debug-build';
+import { V4_SETUP_CLIENTS, V5_SETUP_CLIENTS } from './global-flags';
+import { matchRoutes } from './route-utils';
+import type { MatchPath, RouteConfig, RouterHistory } from './types';
+
+const INTEGRATION_NAME_V4 = 'ReactRouterV4';
+
+const INTEGRATION_NAME_V5 = 'ReactRouterV5';
+
+interface DefaultReactRouterOptions {
+  /**
+   * The history object from `createBrowserHistory` (or equivalent).
+   */
+  history: RouterHistory;
+}
+
+interface RouteConfigReactRouterOptions extends DefaultReactRouterOptions {
+  /**
+   * An array of route configs as per the `react-router-config` library
+   */
+  routes: RouteConfig[];
+  /**
+   * The `matchPath` function from the `react-router` library
+   */
+  matchPath: MatchPath;
+}
+
+/**
+ * Options for React Router v4 and v4 integration
+ */
+type ReactRouterOptions = DefaultReactRouterOptions | RouteConfigReactRouterOptions;
```

Comment:
This union makes the DX waaay nicer, but I don't want to port it to the `routingInstrumentation` in fear of breaking anything.